### PR TITLE
Added `--oauth` option to the `saloon:connector` command

### PR DESCRIPTION
**Pull request by ash-jc-allen**
Hey @Sammyjo20! This PR proposes removing the `saloon:oauth-connector` command for v3 of the plugin and adding a new `--oauth` option to the `saloon:connector` command.

Previously, if you wanted to create an OAuth-ready connector, you could run the following command:

```text
php artisan saloon:oauth-connector GitHub GitHubConnector
```

With the proposed change, you'd be able to run:

```text
php artisan saloon:connector GitHub GitHubConnector --oauth
```

I've also added the ability for the developer to choose whether to use the OAuth boilerplate when using the wizard to create their class. I wasn't too sure if this was something you'd want, but thought it'd be worth proposing seeing as it was only a few lines for me to add. That looks like so:

Screenshot 2023-08-30 at 23 13 45

If this is something you think will be useful, please give me a shout if you'd like anything changing 😄

### CODE REVIEW

**Comment by ash-jc-allen** [comment]
Looking at this, I think there might be a slight oversight on my behalf. For the `select` to work, it will require the `laravel/prompts` package to be installed. I think this was only added in Laravel 10.17, so it'd break for anyone before then. Maybe I should remove that part for now? 🤔

File: src/Console/Commands/MakeConnector.php [hunk 1]
```diff
@@ -34,10 +39,38 @@ class MakeConnector extends MakeCommand
...
+     * @return void
      */
-    protected $stub = 'saloon.connector.stub';
+    protected function afterPromptingForMissingArguments(InputInterface $input, OutputInterface $output)
+    {
+        if ($this->didReceiveOptions($input)) {
+            return;
+        }
+
+        $type = select('Should the connector support OAuth?', [
```
Comment (Sammyjo20) [inline src/Console/Commands/MakeConnector.php]: Hey Ash! Would you mind using the old prompt design to keep Laravel 9 support?

**Review by Sammyjo20 — CHANGES_REQUESTED** [review]
Thanks @ash-jc-allen! Just one small thing to do with prompts!

File: src/Console/Commands/MakeConnector.php (hunk 1 above)
Comment (ash-jc-allen) [reply src/Console/Commands/MakeConnector.php]: Hey Sam! Yeah sure, I'll get this updated now 😄

Comment (ash-jc-allen) [reply src/Console/Commands/MakeConnector.php]: I've added this now, so it _should_ hopefully work with Laravel 9 🤞

**Review by Sammyjo20 — APPROVED** [review]
Thanks Ash!

**Pull request merged**